frontend/internal-packages/agent/src/workflow/nodes/validateInitialSchemaNode.test.ts

```diff
@@ -1,9 +1,262 @@
+          extensions: {},
+        }),
+        analyzedRequirements: {
+          businessRequirement: '',
+          functionalRequirements: {},
+          nonFunctionalRequirements: {},
+        },
+        testcases: [],
+        buildingSchemaId: 'test-building-schema-id',
+        latestVersionNumber: 1,
+        next: 'leadAgent',
+      }
+
+      // Act
+      const result = await graph.invoke(state, config)
+
+      // Assert
+      // Schema validation should succeed with valid schema
+      expect(result).toEqual(state)
+
+      // Should have users table with correct structure
+      expect(result.schemaData.tables['users']).toBeDefined()
+      const usersTable = result.schemaData.tables['users']
+      expect(usersTable?.name).toBe('users')
+      expect(usersTable?.columns['id']?.type).toBe('uuid')
+      expect(usersTable?.columns['email']?.type).toBe('varchar')
+    }, 30000) // 30 second timeout for CI/preview environments
+  })
+
+  describe('Non-first execution scenarios', () => {
+    it('should skip validation when AI messages already exist', async () => {
+      // Arrange
+      const graph = new StateGraph(workflowAnnotation)
+        .addNode('validateInitialSchema', validateInitialSchemaNode)
+        .addEdge(START, 'validateInitialSchema')
+        .addEdge('validateInitialSchema', END)
+        .compile()
+
+      const { config, context } = await getTestConfig()
+
+      const state: WorkflowState = {
+        messages: [
+          new HumanMessage('First message'),
+          new AIMessage('AI response'), // This indicates non-first execution
+          new HumanMessage('Follow-up message'),
+        ],
+        designSessionId: context.designSessionId,
+        organizationId: context.organizationId,
+        userId: context.userId,
+        userInput: 'Follow-up message',
+        schemaData: aSchema({
+          tables: {
+            users: aTable({
+              name: 'users',
+              columns: {
+                id: aColumn({
+                  name: 'id',
+                  type: 'uuid',
+                  notNull: true,
+                }),
+              },
+            }),
+          },
+          enums: {},
+          extensions: {},
+        }),
+        analyzedRequirements: {
+          businessRequirement: '',
+          functionalRequirements: {},
+          nonFunctionalRequirements: {},
+        },
+        testcases: [],
+        buildingSchemaId: 'test-building-schema-id',
+        latestVersionNumber: 1,
+        next: 'leadAgent',
+      }
+
+      // Act
+      const result = await graph.invoke(state, config)
+
+      // Assert
+      // Non-first execution should return state unchanged (validation is skipped)
+      expect(result).toEqual(state)
+      expect(result.userInput).toBe('Follow-up message')
+
+      // Should preserve the existing messages including AI message
+      expect(result.messages).toHaveLength(3)
+      expect(result.messages[1]).toBeInstanceOf(AIMessage)
```

Comment:
Why is this assertion necessary? Isn't it already covered by `expect(result).toEqual(state)`?